lib/tests/test_id_prefix.rs

```diff
@@ -253,3 +253,149 @@ fn test_id_prefix() {
         AmbiguousMatch
     );
 }
+
+#[test]
+fn test_id_prefix_divergent() {
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init_with_backend(TestRepoBackend::Git);
+    let repo = &test_repo.repo;
+    let root_commit_id = repo.store().root_commit_id();
+
+    let mut tx = repo.start_transaction(&settings);
+    let mut create_commit_with_change_id =
+        |parent_id: &CommitId, description: &str, change_id: Option<ChangeId>| {
+            let signature = Signature {
+                name: "Some One".to_string(),
+                email: "some.one@example.com".to_string(),
+                timestamp: Timestamp {
+                    timestamp: MillisSinceEpoch(0),
+                    tz_offset: 0,
+                },
+            };
+            let mut builder = tx
+                .mut_repo()
+                .new_commit(
+                    &settings,
+                    vec![parent_id.clone()],
+                    repo.store().empty_merged_tree_id(),
+                )
+                .set_description(description)
+                .set_author(signature.clone())
+                .set_committer(signature);
+            if let Some(change_id) = change_id {
+                builder = builder.set_change_id(change_id);
+            }
+            builder.write().unwrap()
+        };
+
+    let first_commit = create_commit_with_change_id(root_commit_id, "first", None);
```

Comment:
nit: maybe better to hardcode the change_id instead of relying on the seed?

Reply:
Good point, this simplifies a few things.